File: 10-recursive-combination/recursive_combination.py
def recursive_combination(n: int, k: int) -> int:
    # Base cases
    if k == 0 or k == n:
        return 1
    if k > n:
        return 0
    
    # Recursive case
    return recursive_combination(n-1, k-1) + recursive_combination(n-1, k)

def main():
    # Read input
    with open('input.txt', 'r') as f:
        n, k = map(int, f.read().strip().split())
    
    # Calculate combination
    result = recursive_combination(n, k)
    
    # Write output
    with open('output.txt', 'w') as f:
        f.write(str(result))

if __name__ == "__main__":
    main() 

File: 10-recursive-combination/recursive_combination.rs
use std::io;
use std::fs::File;
use std::io::BufRead;
use std::io::BufReader;

fn recursive_combination(n: i32, k: i32) -> i64 {
    // Base cases
    if k == 0 || k == n {
        return 1;
    }
    if k > n {
        return 0;
    }
    
    // Recursive case
    recursive_combination(n - 1, k - 1) + recursive_combination(n - 1, k)
}

fn main() -> io::Result<()> {
    // Read input 
    let file = File::open("input.txt")?;
    let reader = BufReader::new(file);
    let mut lines = reader.lines();
    
    let n: i32 = lines.next().unwrap()?.trim().parse().unwrap();
    let k: i32 = lines.next().unwrap()?.trim().parse().unwrap();
    
    // Calculate result
    let result = recursive_combination(n, k);
    
    // Write output
    std::fs::write("output.txt", result.to_string())?;

    
    Ok(())
} 